discord/message.py

```diff
@@ -656,6 +658,11 @@ def __init__(
         self.stickers = [Sticker(data=d, state=state) for d in data.get('stickers', [])]
         self.components = [_component_factory(d) for d in data.get('components', [])]
 
+        try:
+            self.guild = channel.guild
+        except AttributeError:
+            self.guild = state._get_guild(utils._get_as_snowflake(data, 'guild_id'))
```

Comment:
```suggestion
        try:
            guild = channel.guild
        except AttributeError:
            guild: Optional[Guild] = state._get_guild(utils._get_as_snowflake(data, 'guild_id'))

        self.guild = guild
```
this should ideally work, assuming a solution for microsoft/pyright#2036 is made.